src/careamics/careamist.py

```diff
@@ -174,9 +181,23 @@ def __init__(
             else:
                 self.model, self.cfg = load_pretrained(source)
 
+        # define in init
+        self.callbacks: list[Callback]
         # define the checkpoint saving callback
         self._define_callbacks(callbacks)
 
+        # TODO: move to method
+        # prediction writer callback reference
+        # placeholder write strategy — will be changed at predict_to_disk call
+        # TODO: use a placeholder write strategy that will raise an error?
+        write_strategy = create_write_strategy("tiff", tiled=False)
+        self.prediction_writer = PredictionWriterCallback(
+            write_strategy=write_strategy, dirpath=self.work_dir / "predictions"
+        )
+        # writing predictions is turned off
+        self.prediction_writer.set_writing_predictions(False)
```

Comment:
Callbacks have to be given to the lightning trainer at initialisation, which is in `CAREamist.__init__`, (lightning does not have any mechanism to add callbacks at a later time). This means without the toggle, predictions will be written every time `trainer.predict` is called, including when it is called in `CAREamist.predict`. Firstly, users may not want to write predictions every time they predict. Secondly, as it is currently implemented, it is not compatible with predicting on arrays, because the prediction writer cannot get the filenames from anywhere.